modules/integration/tests-common/admin-clients/src/main/java/org/wso2/am/admin/clients/mediation/SynapseConfigAdminClient.java

```diff
@@ -0,0 +1,222 @@
+     *
+     * @param serviceName - service name
+     * @throws java.rmi.RemoteException throwable exception
+     */
+    public void deleteConfiguration(String serviceName) throws RemoteException {
+        configServiceAdminStub.deleteConfiguration(serviceName);
+    }
+
+    /**
+     * Get current synapse configuration
+     *
+     * @return synapse configuration
+     * @throws java.rmi.RemoteException throwable exception
+     */
+    public String getConfiguration() throws RemoteException {
+        return configServiceAdminStub.getConfiguration();
+    }
+
+    /**
+     * @return configuration list
+     * @throws java.rmi.RemoteException throwable exception
+     */
+    public ConfigurationInformation[] getConfigurationList() throws RemoteException {
+        return configServiceAdminStub.getConfigurationList();
+    }
+
+    /**
+     * save synapse configuration
+     *
+     * @throws java.rmi.RemoteException throwable exception
+     */
+    public void saveConfigurationToDisk() throws RemoteException {
+        configServiceAdminStub.saveConfigurationToDisk();
+    }
+
+    /**
+     * update synapse configuration
+     *
+     * @param configuration - synapse configuration
+     * @return configuration update status
+     * @throws java.rmi.RemoteException            throwable exception
+     * @throws javax.servlet.ServletException      throwable exception
+     * @throws javax.xml.stream.XMLStreamException throwable exception
+     */
+    public boolean updateConfiguration(String configuration)
+            throws XMLStreamException, ServletException, RemoteException {
+        return configServiceAdminStub.updateConfiguration(createOMElement(configuration));
+    }
+
+    /**
+     * Uploads synapse config bu using a file
+     *
+     * @param file -File that contains the synapse configuration
+     * @return
+     */
+    public boolean updateConfiguration(File file)
+            throws IOException, SAXException, ParserConfigurationException, TransformerException,
+            XMLStreamException, ServletException {
+        boolean success = false;
+        DocumentBuilderFactory factory = DocumentBuilderFactory.newInstance();
+        DocumentBuilder docBuilder = factory.newDocumentBuilder();
+        if (file.exists()) {
+            Document doc = docBuilder.parse(file);
+            String fileContent = getStringFromDocument(doc);
+            success = configServiceAdminStub.updateConfiguration(createOMElement(fileContent));
+        }
+        return success;
+    }
+
+    /**
+     * update synapse configuration
+     *
+     * @param configuration - synapse configuration
+     * @return configuration update status
+     * @throws java.rmi.RemoteException            throwable exception
+     * @throws javax.servlet.ServletException      throwable exception
+     * @throws javax.xml.stream.XMLStreamException throwable exception
+     */
+    public boolean updateConfiguration(OMElement configuration)
```

Comment:
This is a admin client and we need to throw the original excepting to users. As per the offline disscusstion we do not use a custom exception here